src/main/java/server/ClientConnection.java

```diff
@@ -35,9 +27,12 @@ public boolean detectEOF() {
     private boolean EOFReached = false;
 
     private String convertInputStreamToString(InputStream is) throws IOException {
-        BufferedReader bufReader = new BufferedReader(new InputStreamReader(is));
-        String inputStreamContent = bufReader.readLine();
-        EOFReached = inputStreamContent == null;
+        final int RAW_DATA_SIZE = Byte.MAX_VALUE * 100;
+        byte[] inputStreamRawData = new byte[RAW_DATA_SIZE];
```

Comment:
Why do you need to use a byte array instead of a `BufferedReader`?

Reply:
Initially I was using `#readLine()` but when the buffer had multiple lines it only read the first one. I implemented a while loop ( with the break condition `readLine() == null` as that signifies EOF for `BufferedReader`). Instead of reading each line in the input stream, it would block the server waiting for input. 

This caused an infinite loop where it would read one line of user input and immediately asked for more input. Because of this, the program would be stuck in `ClientConnection#readInput()`. It could only be exited by using `CTRL+C`. 

Using the above method, I was able to prompt the user once for input and read the entire stream, including the newlines. From the things I found on stackoverflow, it looks like `BufferedReader#readLine()` is better used for files instead of user input.